src/Controllers/UserController.js

```diff
@@ -60,11 +60,18 @@ export class UserController extends AdaptableController {
       updateFields._email_verify_token_expires_at = {__op: 'Delete'};
     }
 
-    return this.config.database.update('_User', query, updateFields).then((document) => {
-      if (!document) {
-        throw undefined;
-      }
-      return Promise.resolve(document);
+    var self = this;
+    var checkIfAlreadyVerified = new RestQuery(self.config, Auth.master(self.config), '_User', {username: username, emailVerified: true});
```

Comment:
Clarify since thumbs up was ambiguous- can I leave the `resend_verification_email` emailVerified check?

Fixed the verifyEmail return to always return a _User document when not throwing undefined